aicsimageio/transforms.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from collections import Counter
+
+from . import types
+from .dimensions import DEFAULT_DIMENSION_ORDER_LIST
+from .exceptions import ConflictingArgumentsError
+
+###############################################################################
+
+
+def transpose_to_dims(
+    data: types.ArrayLike,
+    given_dims: str,
+    return_dims: str,
+) -> types.ArrayLike:
+    """
+    This shuffles the data dimensions from given_dims to return_dims. Each dimension
+    must be present in given_dims must be used in return_dims
+
+    data: types.ArrayLike
+        Either a dask array or numpy.ndarray of arbitrary shape but with the dimensions
+        specified in given_dims
+    given_dims: str
+        The dimension ordering of data, "CZYX", "VBTCXZY" etc
+    return_dims: str
+        The dimension ordering of the return data
+
+    Returns
+    -------
+    data: types.ArrayLike
+        The data with the specified dimension ordering.
+
+    Raises
+    ------
+    ConflictingArgumentsError: given_dims and return_dims are incompatible.
+    """
+    # Use a counter to track that the contents are composed of the same letters
+    # and that no letter is repeated
+    if (
+        Counter(given_dims) != Counter(return_dims)
+        or max(Counter(given_dims).values()) > 1
+    ):
+        raise ConflictingArgumentsError(
+            f"given_dims={given_dims} and return_dims={return_dims} are incompatible."
+        )
+    # Resort the data into return_dims order
+    match_map = {dim: given_dims.find(dim) for dim in given_dims}
+    transposer = []
+    for dim in return_dims:
+        transposer.append(match_map[dim])
+    data = data.transpose(transposer)
+    return data
+
+
+def reshape_data(
+    data: types.ArrayLike, given_dims: str, return_dims: str, **kwargs
+) -> types.ArrayLike:
+    """
```

Comment:
The functionality here is great. It's grown quite a bit since it's inception. The comments here are good. One request that might already be in place is are there tests that exemplify each of these behaviors so that there are living checks and examples of how it's intended to work?